### DISCUSSION TITLE
Authentication with CRAM-MD5 not working

### DISCUSSION THREAD

**Discussion by Edelwyss** [postalserver/postal#2117] [Help with using Postal] [answered]
I always get the following error message when using CRAM-MD5 for authentication:

> 535 Denied

I tested this with multiple clients (swaks, thunderbird etc.) it fails every time. I works fine with "LOGIN" or "PLAIN". Any known issues with CRAM-MD5 in postal 2.1.1?

**Comment by willpower232** [comment, marked as answer]
The last time someone reported a problem, they weren't using the correct username apparently https://github.com/postalserver/postal/issues/414#issuecomment-401848161

**Reply by Edelwyss** [reply]
The username was indeed wrong. The strange thing is that it worked perfectly fine with the other authentication methods like PLAIN and LOGIN. It seems like PLAIN and LOGIN do not check the username at all. Is this a bug or a feature?

**Reply by willpower232** [reply]
its certainly an interesting question, I guess the API was developed first so usernames weren't a consideration until CRAM-MD5 required it for the hash :sweat_smile:
